lib/cli/src/commands/create_exe.rs

```diff
@@ -344,1043 +256,1624 @@ impl CreateExe {
+                obj.write_stream(&mut writer)
+                    .map_err(|err| anyhow::anyhow!(err.to_string()))?;
+                writer.flush()?;
             }
-
-            #[cfg(feature = "debug")]
-            log::debug!("{:?}", cmd);
-            cmd.output().context("Could not execute `zig`")?
-        };
-        if !compilation.status.success() {
-            return Err(anyhow::anyhow!(String::from_utf8_lossy(
-                &compilation.stderr
-            )
-            .to_string()));
         }
-        Ok(())
     }
 
-    // Write the volumes.o file
-    #[cfg(feature = "webc_runner")]
-    fn write_volume_obj(
-        volume_bytes: &[u8],
-        target: &Target,
-        output_path: &Path,
-    ) -> anyhow::Result<PathBuf> {
-        #[cfg(not(windows))]
-        let volume_object_path = output_path.join("volumes.o");
-        #[cfg(windows)]
-        let volume_object_path = output_path.join("volumes.obj");
-
-        let mut volumes_object = get_object_for_target(target.triple())?;
-        emit_serialized(
-            &mut volumes_object,
-            volume_bytes,
-            target.triple(),
-            "VOLUMES",
-        )?;
-
-        let mut writer = BufWriter::new(File::create(&volume_object_path)?);
-        volumes_object
-            .write_stream(&mut writer)
-            .map_err(|err| anyhow::anyhow!(err.to_string()))?;
-        writer.flush()?;
-        drop(writer);
-
-        Ok(volume_object_path)
+    Ok(module_infos)
+}
+
+/// Compile the C code.
+fn run_c_compile(
+    path_to_c_src: &Path,
+    output_name: &Path,
+    target: &Triple,
+    debug: bool,
+) -> anyhow::Result<()> {
+    #[cfg(not(windows))]
+    let c_compiler = "cc";
+    // We must use a C++ compiler on Windows because wasm.h uses `static_assert`
+    // which isn't available in `clang` on Windows.
+    #[cfg(windows)]
+    let c_compiler = "clang++";
+
+    let mut command = Command::new(c_compiler);
+    let mut command = command
+        .arg("-Wall")
+        .arg("-O2")
+        .arg("-c")
+        .arg(path_to_c_src)
+        .arg("-I")
+        .arg(utils::get_wasmer_include_directory()?);
+
+    // On some compiler -target isn't implemented
+    if *target != Triple::host() {
+        command = command.arg("-target").arg(format!("{}", target));
     }
 
-    #[cfg(feature = "webc_runner")]
-    pub(crate) fn create_objs_pirita(
-        store: &Store,
-        file: &WebCMmap,
-        target: &Target,
-        output_path: &Path,
-        object_format: ObjectFormat,
-    ) -> anyhow::Result<()> {
-        if !output_path.is_dir() {
-            return Err(anyhow::anyhow!(
-                "Expected {} to be an output directory, not a file",
-                output_path.display()
-            ));
-        }
+    let command = command.arg("-o").arg(output_name);
 
-        std::fs::create_dir_all(output_path)?;
-        std::fs::create_dir_all(output_path.join("atoms"))?;
+    if debug {
```

Comment:
No, because the logger isn't enabled by default. If we put `log::debug` it won't be printed out to the command line and if I use `log::debug` it will be interspersed with other irrelevant debug output. `debug` is set if `debug_dir.is_some()`, so there is a point to this.